sycl/doc/EnvironmentVariables.md

```diff
@@ -84,9 +84,22 @@ The following examples further illustrate the usage of this environment variable
 Notes:
 - The backend argument is always required. An error will be thrown if it is absent.
 - Additionally, the backend MUST be followed by colon ( `:` ) and at least one device specifier of some sort, else an error is thrown.
-- For sub-devices and sub-sub-devices, the parent device must support partitioning (`info::partition_property::partition_by_affinity_domain` and `info::partition_affinity_domain::next_partitionable`. See the SYCL 2020 specification for a precise definition.) For Intel GPUs, the sub-device and sub-sub-device syntax can be used to expose tiles or CCSs to the SYCL application as root devices.  The exact mapping between sub-device, sub-sub-device, tiles, and CCSs is specific to the hardware.
+- The sub-device and sub-sub-device syntax attempt to partition the root device
+  according to the rules defined by
+  `info::partition_property::partition_by_affinity_domain` and
+  `info::partition_affinity_domain::next_partitionable`.
+  (See the SYCL 2020 specification for a precise definition.)
+  The root device is determined by the underlying backend.
+- When using the Level Zero or OpenCL backends, see also the documentation of
+  the [`ZE_FLAT_DEVICE_HIERARCHY`][ze-env] environment variable because it
+  affects how these backends expose root devices to SYCL.
```

Comment:
https://spec.oneapi.io/level-zero/latest/core/PROG.html#environment-variables says

> Defines device hierarchy model exposed by Level Zero driver implementation

`ZE_FLAT_DEVICE_HIERARCHY` should not affect OpenCL backend. Right?

Reply:
I thought @jandres742 told me that it affected both Level Zero and OpenCL.  Is that correct, Jaime?  Do we document this someplace?

Reply:
Yes, I suppose so.  I think we could changed the wording to one of the following depending on @jandres742's response:

> When using the Level Zero backend ...

> When using the Level Zero or Intel OpenCL GPU backend ...

> When using backends from Intel ...

Reply:
OpenCL GPU driver uses the same environment variable, but just because Intel L0 and OCL GPU drivers come from the same codebase. However, as mentioned above, that is weird, as ZE_FLAT_DEVICE_HIERARCHY  is not a OCL spec variable. So in that sense, and considering anyway that L0 is the official backend for oneAPI, I would just not mention it.

Reply:
Updated in 7a1971c370fe172da96180d1d0009eecc64f1876 to say it applies only to Level Zero.